### ISSUE TITLE
join: array of objects

### ISSUE THREAD

**Issue by matidio** [stedolan/jq#544]
Hello,

first, sorry that use the issue board for a support request. I don't know any forum for asking JQ related questions.

I have the following input:

```{
    "directors": [
        {
            "name": "Quentin Tarantino",
            "nameId": "nm0000233"
        },
        {
            "name": "Quentin Tarantino (test)",
            "nameId": "nm0000234"
        }
    ],
    "languages": [
        "English",
        "German",
        "French"
    ]
}```

If I apply the following filter on the input above
`"Languages: \(.languages | join(", "))"`
 I get the wished output:
`"Languages: English, German, French"`

When I use instead the following filter on the same input
`"Directors: \(.directors[].name | join(", "))"`
the following error is shown:
`jq: error: Cannot iterate over string`

How do I have to approach an array of objects with the join function to get a similar result with an array of strings like the languages example above?

I read the manual but I'm very new to `JQ` so please be patient.

A second question in this thread to not open another issue. I would like to use the same file with different filters to apply to a json file (input) but I'm not sure how I can apply various filters to the same input. I.e., I would like to use first the above languages filter and after that the directors filter. How can I do this with the same file so that I have to start JQ only once and not for every filter line once.

My filter file should look like this:
```"Languages: \(.languages | join(", "))"```
```"Directors: \(.directors[].name | join(", "))"```

Every filter line should be applied to the input (which is, btw, also a file)

Thanks in advance for your help.

Matidio
